src/tracks/PatternTrack.cpp

```diff
@@ -109,18 +109,26 @@ bool PatternTrack::play( const TimePos & _start, const fpp_t _frames,
 
 	TimePos lastPosition;
 	TimePos lastLen;
+	tick_t lastOffset = 0;
 	for (const auto& clip : clips)
 	{
 		if (!clip->isMuted() && clip->startPosition() >= lastPosition)
 		{
 			lastPosition = clip->startPosition();
 			lastLen = clip->length();
+			tick_t patLen = Engine::patternStore()->lengthOfPattern(static_cast<PatternClip*>(clip)->patternIndex())
```

Comment:
And again with `patLength`, or even better `patternLength`. :)

Reply:
To be honest, when I write code, I normally always type things out, even if it gets like 20 characters wide. As this is my first contribution here, I tried to keep the style but if it is okay, I would be glad using more transparent namings in the future.